src/storage/txn/commands/compare_and_set.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::storage::kv::{Modify, WriteData};
+use crate::storage::lock_manager::LockManager;
+use crate::storage::raw;
+use crate::storage::txn::commands::{
+    Command, CommandExt, ResponsePolicy, TypedCommand, WriteCommand, WriteContext, WriteResult,
+};
+use crate::storage::txn::Result;
+use crate::storage::{ProcessResult, Snapshot};
+use engine_traits::CfName;
+use txn_types::{Key, Value};
+
+command! {
+    /// CompareAndSet to check whether the previous value of the key equals to the given value.
+    /// If they are equal, write the new value, otherwise return the previous value.
+    AtomicCompareAndSet:
+        cmd_ty => Option<Value>,
+        display => "kv::command::atomic_compare_and_set {:?}", (ctx),
+        content => {
+            cf: CfName,
+            key: Key,
+            previous_value: Option<Value>,
+            value: Value,
+            enable_ttl: bool,
```

Comment:
how about merging `enable_ttl` and `ttl` into
```
ttl: Option<u64>
```

Reply:
Done...